garden_ai/app/create.py

```diff
@@ -0,0 +1,172 @@
+    return name.strip() if name else ""
+
+
+# @app.callback()
+def create_garden(
+    directory: Path = typer.Argument(
+        None,
+        callback=setup_directory,
+        dir_okay=True,
+        file_okay=False,
+        writable=True,
+        readable=True,
+        resolve_path=True,
+        help=(
+            "(Optional) if specified, this generates a directory with subfolders to help organize the new Garden. "
+            "This is most likely to be useful if you want to track your Garden/Pipeline development with GitHub."
+        ),
+    ),
+    authors: List[str] = typer.Option(
+        None,
+        "-a",
+        "--author",
+        help=(
+            "Name an author of this Garden. Repeat this to indicate multiple authors: "
+            "`garden create ... --author='Mendel, Gregor' --author 'Other-Author, Anne' ...` (order is preserved)."
+        ),
+        rich_help_panel="Required",
+        prompt=False,  # NOTE: automatic prompting won't play nice with list values
+    ),
+    title: str = typer.Option(
+        ...,
+        "-t",
+        "--title",
+        prompt="Please enter a title for your Garden",
+        help="Provide an official title (as it should appear in citations)",
+        rich_help_panel="Required",
+    ),
+    year: str = typer.Option(
+        str(datetime.now().year),  # default to current year
+        "-y",
+        "--year",
+        rich_help_panel="Required",
+    ),
+    contributors: List[str] = typer.Option(
+        None,
+        "-c",
+        "--contributor",
+        help=(
+            "Acknowledge a contributor in this Garden. Repeat to indicate multiple (like --author). "
+        ),
+        rich_help_panel="Recommended",
+    ),
+    description: Optional[str] = typer.Option(
+        None,
+        "-d",
+        "--description",
+        help=(
+            "A brief summary of the Garden and/or its purpose, to aid discovery by other Gardeners."
+        ),
+        rich_help_panel="Recommended",
+    ),
+    tags: List[str] = typer.Option(
+        None,
+        "--tag",
+        help="Add a tag, keyword, key phrase or other classification pertaining to the Garden.",
+        rich_help_panel="Recommended",
+    ),
+    verbose: bool = typer.Option(
+        False, help="If true, pretty-print Garden's metadata when created."
+    ),
+):
+    """Create a new Garden"""
+    while not authors:
+        # repeatedly prompt for at least one author until one is given
+        name = validate_name(Prompt.ask("Please enter at least one author (required)"))
+        if not name:
+            continue
+
+        authors = [name]
+        # prompt for additional authors until one is *not* given
+        while True:
+            name = validate_name(
+                Prompt.ask("Add another author? (leave blank to finish)")
+            )
+            if name:
+                authors += [name]
+            else:
+                break
```

Comment:
this should be simplified to provide a list of comma-separated authors (in my opinion at least). It's less code and it's more clear.

Reply:
I waffled on this too, but the problem is that people are likely to comma-separate their names as "Lastname, Firstname", so we'd need a different separator. My next thought would be semicolons, but that could be easy to forget or mistake in light of other conventions like `$PATH` being colon-separated etc. Figured that  `--author "lastname, firstname" --author "Familyname, Givenname"` would be more verbose but impossible to accidentally chop someone's name in half when parsing the input or something

Reply:
(though course I should be careful when I say "impossible" https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)